info.limpet.rcp/src/info/limpet/rcp/editors/DataManagerEditor.java

```diff
@@ -100,7 +118,117 @@ public void changed()
 	private Action createCourse;
 	private Action createLocation;
 	private IContext _context = new RCPContext();
+	
+	private IResourceChangeListener resourceChangeListener = new IResourceChangeListener()
+	{
+
+		@Override
+		public void resourceChanged(IResourceChangeEvent event)
+		{
+			IResourceDelta delta = event.getDelta();
+			final int eventType = event.getType();
+			if (delta != null)
+			{
+				try
+				{
+					delta.accept(new IResourceDeltaVisitor()
+					{
+
+						@Override
+						public boolean visit(IResourceDelta delta) throws CoreException
+						{
+							IResource resource = delta.getResource();
+							if (resource instanceof IWorkspaceRoot)
+							{
+								return true;
+							}
+							if (resource instanceof IProject)
+							{
+								IEditorInput input = getEditorInput();
+								if (input instanceof IFileEditorInput)
+								{
+									IProject project = ((IFileEditorInput) input).getFile()
+											.getProject();
+									if (resource.equals(project)
+											&& (eventType == IResourceChangeEvent.PRE_DELETE || eventType == IResourceChangeEvent.PRE_CLOSE))
+									{
+										closeEditor();
+										return false;
+									}
+								}
+								return true;
+							}
+							if (resource instanceof IFolder)
+							{
+								return true;
+							}
+							if (resource instanceof IFile)
+							{
+								IEditorInput input = getEditorInput();
+								if (input instanceof IFileEditorInput)
+								{
+									IFile file = ((IFileEditorInput) input).getFile();
+									if (resource.equals(file)
+											&& delta.getKind() == IResourceDelta.REMOVED)
+									{
+										IPath movedToPath = delta.getMovedToPath();
+										if (movedToPath != null)
+										{
+											IResource path = ResourcesPlugin.getWorkspace().getRoot()
+													.findMember(movedToPath);
+											if (path instanceof IFile)
+											{
+												final FileEditorInput newInput = new FileEditorInput(
+														(IFile) path);
+												Display.getDefault().asyncExec(new Runnable()
+												{
+
+													@Override
+													public void run()
+													{
+														setInputWithNotify(newInput);
+														setPartName(newInput.getName());
+													}
+												});
+											}
+										}
+										else
+										{
+											closeEditor();
+										}
+									} 
+									if (resource.equals(file) && 
+											(delta.getKind() == IResourceDelta.CHANGED && (delta.getFlags() & IResourceDelta.CONTENT) != 0))
+									{
+										// TODO reload
```

Comment:
When a user change lap file contents out of limpet editor we should reload editor's contents. It can be done in another ticket.